### PR TITLE
Server-sign messages

### PR DESCRIPTION
**Pull request by jlbribeiro**
This PR aims to tackle #2 by implementing server-signed messages. While I made my best effort to follow the RFCs ([RFC 1847](https://tools.ietf.org/html/rfc1847) and [RFC 3156](https://tools.ietf.org/html/rfc3156)) guidelines the best I could so this would be compliant, it seems only some mail clients (`mutt` and `Evolution`) are able to correctly display the message as it is currently constructed (as a `multipart/signed` with a `multipart/encrypted` inside); reportedly Thunderbird was not able to open it (it hangs while loading the message). As such, it is being introduced as a per-user feature toggle, so users may turn it on if their client supports it.

I should remind our approach requires us to **encrypt-then-sign** since the payload reaches the server already encrypted on the client-side; from what I could understand most clients will implement **sign-then-encrypt** (see [this comment](http://www.metzdowd.com/pipermail/cryptography/2014-January/019754.html)).

In the future I plan to improve this and expand the feature's mail client support by exploring other options.

Sidenote: the checkbox on the user settings menu is a little off; I believe we can merge this as it is and enhance this on a future design iteration, but I may fix it if necessary for this to be merged.

### CODE REVIEW

File: boxes/tasks.py [hunk 1]
```diff
@@ -2,20 +2,34 @@
 from django.core.mail import EmailMultiAlternatives
 from django.conf import settings
 from django.utils import timezone
+
 from .models import Message
+from .email import GPGSignedEncryptedMessage
 
 from celery import shared_task
 
+import gnupg
```
Comment (dethos) [inline boxes/tasks.py]: This import seems to be unnecessary.

File: boxes/tasks.py [hunk 2]
```diff
@@ -2,20 +2,34 @@
...
-    email = EmailMultiAlternatives(subject, form_data["message"],
-                                   settings.DEFAULT_FROM_EMAIL,
-                                   [box.owner.email])
+
+    if box.owner.server_signed:
+      email = GPGSignedEncryptedMessage(subject, msg,
+                                      settings.DEFAULT_FROM_EMAIL,
+                                      [box.owner.email])
+    else:
+      email = EmailMultiAlternatives(subject, msg,
```
Comment (dethos) [inline boxes/tasks.py]: Both lines where `email` is assigned, do not respect the 4 indentation spaces, used on the remaining of the project.

**Comment by dethos** [comment]
@jlbribeiro Just 2 minor notes, the remaining looks good to me. When they are fixed, I will happily merge the PR.

File: boxes/tasks.py (hunk 1 above)
Comment (jlbribeiro) [inline boxes/tasks.py]: @dethos: Well noticed, removing.

File: boxes/tasks.py (hunk 2 above)
Comment (jlbribeiro) [inline boxes/tasks.py]: @dethos: Nice catch, also fixed.

**Comment by jlbribeiro** [comment]
@dethos: fixed your minors, also fixed the digest algo thing you mentioned. Give it a look and tell me if it is good to go.

**Pull request merged**